dspace-rest/src/main/java/org/dspace/rest/ItemsResource.java

```diff
@@ -872,11 +849,11 @@ else if (!AuthorizeServiceImpl.authorizeActionBoolean(context, bitstream, org.ds
             log.trace("Deleting bitstream...");
             for (org.dspace.content.Bundle bundle : item.getBundles())
             {
-                for (org.dspace.content.Bitstream bit : bundle.getBitstreams())
+                for (org.dspace.content.BundleBitstream bit : bundle.getBitstreams())
                 {
-                    if (bit == bitstream)
+                    if (bit.getBitstream() == bitstream)
                     {
-                        bundle.removeBitstream(bitstream);
+                        bundleService.removeBitstream(context, bundle, bitstream);
```

Comment:
Removing a bitstream from a bundle while looping will cause ConcurrentModificationException, best to use an iterator & remove the bitstream from the iterator prior to removing from the bundle.